### PR TITLE
feat: database connection

### PR DESCRIPTION
**Pull request by totorototo**
do not merge yet

### CODE REVIEW

File: app/store/sagas/appSagas/login.js [hunk 1]
```diff
@@ -49,12 +50,15 @@ export function* authenticationFlowSaga() {
         // eslint-disable-next-line no-unused-vars, prefer-const
         let { token, error } = yield call(authorize, navigation.params.code);
         if (!error) {
+          const task = yield spawn(watchDatabase);
```
Comment (guipasmoi) [inline app/store/sagas/appSagas/login.js]: why swpawn and not fork?

Comment (guipasmoi) [reply app/store/sagas/appSagas/login.js]: this shouldn t be here. this has nothing to do with login with strava 
watch for a successfull login et logout to start watchdatabase

File: app/store/sagas/race.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { call, cancelled, take } from "redux-saga/effects";
```
Comment (guipasmoi) [inline app/store/sagas/race.js]: there is somthing wrong with the name  app/store/sagas/race.js and what is inside

File: app/store/services/database.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export const registerCallBackForEvent = cb => {
+  database.registerForEvent(cb);
+};
+
+export const unregisterCallBackForEvent = cb => {
+  database.unregisterForEvent(cb);
+};
+
+export const writeData = (data = {}) => database.write(data);
```
Comment (guipasmoi) [inline app/store/services/database.js]: it s not key - data ?

File: app/store/sagas/race.js (hunk 2 above)
Comment (totorototo) [reply app/store/sagas/race.js]: to be renamed! I did not find the right name for it yet

File: app/store/sagas/appSagas/login.js (hunk 1 above)
Comment (totorototo) [reply app/store/sagas/appSagas/login.js]: 👍

Comment (totorototo) [reply app/store/sagas/appSagas/login.js]: this was an temporary implementation. 
Has to be cleaned up.

File: app/store/services/database.js (hunk 3 above)
Comment (totorototo) [reply app/store/services/database.js]: yep!

File: app/store/sagas/location.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  });
+}
+
+function* watchDatabase() {
+  const channel = yield call(subscribe);
+
+  try {
+    // eslint-disable no-constant-condition
+    let userSignedOut;
+    while (!userSignedOut) {
```
Comment (guipasmoi) [inline app/store/sagas/location.js]: if you lost connection it looks like its over. no?

File: app/store/services/database.js [hunk 5]
```diff
@@ -16,14 +16,14 @@ export const disconnect = () => {
   database.disconnect();
 };
 
-export const registerCallBackForEvent = cb => {
-  database.registerForEvent(cb);
+export const watchData = cb => {
```
Comment (guipasmoi) [inline app/store/services/database.js]: name should be onChange and it should return a subscription and this subscription should allow you to stop watch

File: app/store/services/database.js [hunk 6]
```diff
@@ -16,14 +16,14 @@ export const disconnect = () => {
...
+  database.registerWatcher(cb);
 };
 
-export const unregisterCallBackForEvent = cb => {
-  database.unregisterForEvent(cb);
+export const unwatchData = cb => {
```
Comment (guipasmoi) [inline app/store/services/database.js]: and this wouldnt be here

**Review by guipasmoi — COMMENTED** [review]
thats better 👍

File: app/store/sagas/location.js (hunk 4 above)
Comment (totorototo) [reply app/store/sagas/location.js]: https://firebase.google.com/docs/database/web/offline-capabilities
we could enable offline capabilities! 
I have to deal with that later! How to handle offline!!!

Comment (guipasmoi) [reply app/store/sagas/location.js]: i was just saying if you lost the connection you have to restart the app

File: app/store/sagas/location.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import { call, cancelled, take, takeEvery, race } from "redux-saga/effects";
```
Comment (guipasmoi) [inline app/store/sagas/location.js]: the filename   app/store/sagas/location.js should be  app/store/sagas/remoteData?
i see nothing location spécific

File: app/store/services/database.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import Config from "react-native-config";
+import { database } from "./helpers/database";
+
+export const authenticate = () => {
+  const configuration = {
```
Comment (guipasmoi) [inline app/store/services/database.js]: we should try to make it stateless

Comment (totorototo) [reply app/store/services/database.js]: I don't get your point!

File: app/routes/screens/race/Race.js [hunk 9]
```diff
@@ -103,12 +105,20 @@ class RacePredictor extends Component {
               <MapView.Marker
                 coordinate={location.coordinates}
                 title={id}
-                description={location.time}
-                pinColor="#004C02"
+                description={new Date(location.time).toLocaleString()}
```
Comment (guipasmoi) [inline app/routes/screens/race/Race.js]: extract to format time?

File: app/routes/screens/race/Race.js [hunk 10]
```diff
@@ -55,8 +55,10 @@ class RacePredictor extends Component {
     }).isRequired
   };
 
-  static finish() {
-    console.log("Countdown finished");
+  static intToColor(id = 0) {
```
Comment (guipasmoi) [inline app/routes/screens/race/Race.js]: what color for what

**Review by guipasmoi — APPROVED** [review]

File: app/routes/screens/race/Race.js (hunk 10 above)
Comment (totorototo) [reply app/routes/screens/race/Race.js]: one color to rule them all! 🤐 
helper function to convert int (athlete ID) to color (Map view marker color). 
So in the end, each athlete could be  easily spotted on da map! 
Nice, isn't it?

File: app/routes/screens/race/Race.js (hunk 9 above)
Comment (totorototo) [reply app/routes/screens/race/Race.js]: ?

Comment (guipasmoi) [reply app/routes/screens/race/Race.js]: const formatDate =(timestamp) =>new Date(timestamp).toLocaleString()

File: app/routes/screens/race/Race.js (hunk 10 above)
Comment (guipasmoi) [reply app/routes/screens/race/Race.js]: intocolort is an helper it should be sharable

**Pull request merged**